Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/LoadTestDataPatientBean.java

```diff
@@ -83,82 +113,246 @@ public void setSelectedPatient(Patient selectedPatient) {
+            return loadTestDataService.deletePhonenumber(selectedPhonenumber);
+        } else {
+            addPatientErrorMessages("Phonenumber have to be select to be deleted");
+        }
+        return false;
+    }
+
+    public boolean savePhonenumber() {
+        boolean actionResult = false;
+
+        if (isValidPatientId()) {
+            try {
+                Phonenumber phonenumber = readPhonenumber();
+                phonenumber.setPatient(withPatient);
+                actionResult = loadTestDataService.savePhonenumber(phonenumber);
+
+                if (actionResult) {
+                    addPatientInfoMessages("Save Phonenumber Successful: " + phonenumber.getPhonenumberId());
+                    clearInputsPhonenumbers();
+                }
+            } catch (LoadTestDataException e) {
+                logPatientError("Patient Phonenumber", e);
+            }
+        } else {
+            addPatientErrorMessages("Phonenumber cannot be save: patient does not exist.");
+        }
+        return actionResult;
+    }
+
+    public List<Address> getAddresses() {
+        if (isValidPatientId()) {
+            return loadTestDataService.getAllAddressesBy(getPatientId());
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    public boolean deleteAddress() {
+        if (selectedAddress != null) {
+            return loadTestDataService.deleteAddress(selectedAddress);
+        } else {
+            addPatientErrorMessages("Address have to be select to be deleted");
+        }
+        return false;
+    }
+
+    public boolean saveAddress() {
+        boolean actionResult = false;
+        if (isValidPatientId()) {
+            try {
+                Address address = readAddress();
+                address.setPatient(withPatient);
+                actionResult = loadTestDataService.saveAddress(address);
+
+                if (actionResult) {
+                    addPatientInfoMessages("Save Address Successful: " + address.getAddressId());
+                    clearInputsAddresses();
+                }
+            } catch (LoadTestDataException e) {
+                logPatientError("Patient Address", e);
+            }
+        } else {
+            addPatientErrorMessages("Address cannot be save: patient does not exist.");
+        }
+
+        return actionResult;
+    }
+
+    public List<Identifier> getIdentifiers() {
+        if (isValidPatientId()) {
+            return loadTestDataService.getAllIdentiersBy(getPatientId());
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    public boolean deleteIdentifier() {
+        if (selectedIdentifier != null) {
+            return loadTestDataService.deleteIdentifier(selectedIdentifier);
+        } else {
+            addPatientErrorMessages("Identifer have to be select to be deleted");
+        }
+        return false;
+    }
+
+    public boolean saveIdentifier() {
+        boolean actionResult = false;
```

Comment:
Refactor the save code to GenericDAO to remove the redundant logic.

Reply:
the save method cannot be refractor